docs/src/design/proposals/APIRedesign.md

```diff
@@ -0,0 +1,1050 @@
+  non-Angular version.)
+* Expected reduction of test scope will also be invalidated by an
+  expected end-of-life.
+
+Other problems:
+
+* Hinders integrating non-Angular components. (Need to wrap with
+  Angular API, e.g. as directives, which may be non-trivial.)
+* Interferes with debugging by swallowing or obscuring exceptions.
+
+# Feedback Review
+
+## Problem Summary
+
+The following attributes of the current API are undesirable:
+
+- [ ] It is difficult to tell "where things are" in the code base.
+- [ ] It is difficult to see how objects are passed around at run-time.
+- [ ] It is difficult to trace flow of control generally.
+- [ ] Multiple interfaces for related concepts (e.g. telemetry) is confusing.
+- [ ] API documentation is missing or not well-formatted for use.
+- [ ] High-level separation of concerns is not made clear.
+- [ ] Interface depth of telemetry API is excessive (esp. `TelemetrySeries`)
+- [ ] Capabilities as a concept lack clarity.
+- [ ] Too many interfaces and concepts to learn.
+- [ ] Exposing third-party APIs (e.g. Angular's) increases the learning curve.
+- [ ] Want more examples, easier-to-use documentation.
+- [ ] UI-relevant features (HTML, CSS) under-documented
+- [ ] Good MVC for views of domain objects not enforced (e.g. plots)
+
+## Positive Features
+
+It is desirable to retain the following features in an API redesign:
+
+- [ ] Creating new features and implementing them additively is well-supported.
+- [ ] Easy to add/remove features which involve multiple concerns.
+- [ ] Features can be self-contained.
+- [ ] Declarative syntax makes it easy to tell what's in use.
+
+## Requirements
+
+The following are considered "must-haves" of any complete API
+redesign:
+
+- [ ] Don't require usage of Angular API.
+- [ ] Don't require support for Angular API.
+
+# Proposals
+
+## RequireJS as dependency injector
+
+Use Require.JS for dependency injection.
+
+Dependencies will then be responsible for being sufficiently
+mutable/extensible/customizable. This can be facilitated by
+adding platform classes which can facilitate the addition
+of reusable components.
+
+Things that we usefully acquire via dependency injection currently:
+
+* Services.
+* Extensions (by category).
+* Configuration constants.
+
+Services would be defined (by whatever component is responsible
+for declaring it) using `define` and the explicit name of the
+service. To allow for the power of composite services, the
+platform would provide a `CompositeService` class that supports
+this process by providing `register`, `decorate`, and `composite`
+methods to register providers, decorators, and aggregators
+respectively. (Note that nomenclature changes are also implied
+here, to map more clearly to the Composite Pattern and to
+avoid the use of the word "provider", which has ambiguity with
+Angular.)
+
+```js
+define(
+  "typeService",
+  ["CompositeService"],
+  function (CompositeService) {
+    var typeService = new CompositeService([
```

Comment:
To me, Angular's DI mechanism is actually preferable here. Angular provides me with an instance of the dependency, which means
a) I don't have to know how to initialize it, Angular has done it for me
b) I don't have to write initialization boilerplate in every place that I use the dependency.
c) I can easily mock and test it because instances are injected.